src/syscall_mocking/mod.rs

```diff
@@ -45,33 +45,42 @@ impl Tracer {
         }
     }
 
-    fn execve(executable: &Path, args: Vec<String>, env: HashMap<String, String>) -> R<()> {
+    fn format_execve_error(executable: &Executable, args: &[String], error: nix::Error) -> String {
+        let (program, hint): (String, String) = match executable {
+            Executable::Program(program) => {
+                let hint = parse_shebang(program)
+                    .map(|shebang| format!("Does \"{}\" exist?", shebang))
+                    .unwrap_or_else(|| "Does your interpreter exist?".to_string());
+                (program.to_string_lossy().to_string(), hint)
+            }
+            Executable::Interpreter(_) => {
+                let hint = format!("Does \"{}\" exist?", executable.to_string_lossy());
+                (args.first().unwrap().to_string(), hint)
+            }
+        };
+        format!(
+            "execve'ing {} failed with error: {}\n{}",
+            program, error, hint,
+        )
+    }
+
+    fn execve(executable: &Executable, args: Vec<String>, env: HashMap<String, String>) -> R<()> {
         let path = CString::new(executable.as_os_str().as_bytes())?;
         let mut c_args = vec![path.clone()];
-        for arg in args {
-            c_args.push(CString::new(arg)?);
+        for arg in &args {
+            c_args.push(CString::new(arg.clone())?);
         }
         let mut c_env = vec![];
         for (key, value) in env {
             c_env.push(CString::new(format!("{}={}", key, value))?);
         }
-        execve(&path, &c_args, &c_env).map_err(|error| {
-            let hint = match parse_shebang(executable) {
-                Some(shebang) => format!("Does \"{}\" exist?", shebang),
-                None => "Does your interpreter exist?".to_string(),
-            };
-            format!(
-                "execve'ing {} failed with error: {}\n{}",
-                executable.to_string_lossy(),
-                error,
-                hint,
-            )
-        })?;
+        execve(&path, &c_args, &c_env)
+            .map_err(|error| Self::format_execve_error(executable, &args, error))?;
```

Comment:
The closure for formatting an error given by `execve` has been outlined into a function above.